pkg/cmd/utils.go

```diff
@@ -298,3 +259,86 @@ func BuildConfig(master, kubeConfig string, f client.Factory) (*rest.Config, err
 	}
 	return config, nil
 }
+
+func GetCompatibleRepoAndTagFromPluginImage(kubeClient kubernetes.Interface, namespace string, targetContainer string) (string, error) {
+	deployment, err := kubeClient.AppsV1().Deployments(namespace).Get(context.TODO(), constants.VeleroDeployment, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Errorf("Failed to get velero deployment in namespace %s", namespace)
+	}
+
+	var repo, tag, image string
+	for _, container := range deployment.Spec.Template.Spec.InitContainers {
+		imageContainer := utils.GetComponentFromImage(container.Image, constants.ImageContainerComponent)
+		if imageContainer == constants.VeleroPluginForVsphere {
+			image = container.Image
+			repo = utils.GetComponentFromImage(image, constants.ImageRepositoryComponent)
+			tag = utils.GetComponentFromImage(image, constants.ImageVersionComponent)
+			break
+		}
+	}
+
+	if image == "" {
+		return "", errors.New("The plugin, velero-plugin-for-vsphere, was not added as an init container of Velero deployment")
+	}
+
+	resultImage := targetContainer
+	if repo != "" {
+		resultImage = repo + "/" + resultImage
+	}
+	if tag != "" {
+		resultImage = resultImage + ":" + tag
+	}
+	return resultImage, nil
+}
+
+func CheckVSphereCSIDriverVersion(kubeClient kubernetes.Interface) error {
+	isCSIInstalled, isVersionOk, err := CheckCSIInstalled(kubeClient)
+	if err != nil {
+		fmt.Println("CSI driver check failed")
+		isCSIInstalled = false
+		isVersionOk = false
+	}
+
+	if !isCSIInstalled {
+		fmt.Println("Velero Plug-in for vSphere requires vSphere CSI/CNS and vSphere 6.7U3 to function. Please install the vSphere CSI/CNS driver")
+	}
+
+	if !isVersionOk {
+		fmt.Printf("vSphere CSI driver version is prior to %s. Velero Plug-in for vSphere requires CSI driver version to be %s or above\n", constants.CsiMinVersion, constants.CsiMinVersion)
+	}
+
+	return err
+}
+
+func CheckVeleroVersion(kubeClient kubernetes.Interface, ns string) error {
+	veleroVersion, err := GetVeleroVersion(kubeClient, ns)
+	if err != nil || veleroVersion == "" {
+		fmt.Println("Failed to get velero version.")
+	} else {
+		if CompareVersion(veleroVersion, constants.VeleroMinVersion) == -1 {
+			fmt.Printf("WARNING: Velero version %s is prior to %s. Velero Plug-in for vSphere requires velero version to be %s or above.\n", veleroVersion, constants.VeleroMinVersion, constants.VeleroMinVersion)
+		}
+	}
+
+	return nil
+}
+
+func CheckPluginImageRepo(kubeClient kubernetes.Interface, ns string, defaultImage string, serverType string) (string, error) {
+	resultImage, err := func(kubeClient kubernetes.Interface) (string, error) {
+		resultImage, err := GetCompatibleRepoAndTagFromPluginImage(kubeClient, ns, serverType)
+		if err != nil {
+			return "", err
+		}
+
+		return resultImage, nil
+	}(kubeClient)
```

Comment:
Why the anonymous function?

Reply:
Just tried to keep the original behavior. 
In the change, we move the block of error checking inside the implementation of `CheckPluginImageRepo`. However, since the original code doesn't error out if `err != nil` but only printing out the error message. To replicate this, I use the anonymous function.

Reply:
You can achieve the same behavior without the anonymous function.

Reply:
> You can achieve the same behavior without the anonymous function.

Good point. Done.